### PR TITLE
light-verifier-sdk: Refactor of arguments

### PR DESCRIPTION
**Pull request by vadorovsky**
Split the big `Transaction` struct into multiple and consolidate transaction inputs (like proofs or amounts) together.

### CODE REVIEW

File: light-system-programs/programs/verifier_program_storage/src/lib.rs [hunk 1]
```diff
@@ -90,20 +90,24 @@ pub mod verifier_program_storage {
...
+        };
 
         process_shielded_transfer_2_in_2_out(
             &ctx,
-            Some(&message_hash),
-            Some(message),
-            &inputs.proof_a,
-            &inputs.proof_b,
-            &inputs.proof_c,
-            &[0u8; 32], // Verifier storage does not support SPL tokens.
```
Comment (ananas-block) [inline light-system-programs/programs/verifier_program_storage/src/lib.rs:103]: Probably makes sense to keep the comment, I was a bit confused at first.

File: light-system-programs/programs/verifier_program_storage/src/lib.rs [hunk 2]
```diff
@@ -90,20 +90,24 @@ pub mod verifier_program_storage {
...
             )?;
-        let message = &ctx.accounts.verifier_state.msg;
-        let message_hash = hash(message).to_bytes();
+        let message = Message::new(&ctx.accounts.verifier_state.msg);
+        let proof = Proof {
+            a: inputs.proof_a,
+            b: inputs.proof_b,
+            c: inputs.proof_c,
+        };
+        let public_amount = Amount {
```
Comment (ananas-block) [inline light-system-programs/programs/verifier_program_storage/src/lib.rs:99]: Did you consider calling it amounts?

File: light-verifier-sdk/src/light_transaction.rs [hunk 3]
```diff
@@ -46,87 +46,73 @@ pub trait Config {
...
+
+pub struct Message<'a> {
+    pub content: &'a Vec<u8>,
+    pub hash: [u8; 32],
+}
+
+impl<'a> Message<'a> {
+    pub fn new(content: &'a Vec<u8>) -> Self {
+        let hash = hash::hash(content).to_bytes();
+        Message { hash, content }
```
Comment (ananas-block) [inline light-verifier-sdk/src/light_transaction.rs:75]: nice!

**Review by ananas-block — COMMENTED** [review]
Looks good, I like it! Just two tiny things.

File: light-system-programs/programs/verifier_program_storage/src/lib.rs (hunk 1 above)
Comment (vadorovsky) [reply light-system-programs/programs/verifier_program_storage/src/lib.rs:103]: Added it back

**Review by ananas-block — APPROVED** [review]

**Pull request merged**